source/WebApp/build.ts

```diff
@@ -81,25 +87,40 @@ const ts = task('ts', async () => {
     ]
 });
 
-const favicons = task('favicons', async () => {
+const icons = task('icons', async () => {
     await jetpack.dirAsync(outputRoot);
-    const pngGeneration = [16, 32, 64, 96, 128, 196, 256].map(size => {
+    const pngGeneration = iconSizes.map(size => {
         // https://github.com/lovell/sharp/issues/729
         const density = size > 128 ? Math.round(72 * size / 128) : 72;
-        return sharp(paths.from.favicon, { density })
+        return sharp(paths.from.icon, { density })
             .resize(size, size)
             .png()
-            .toFile(paths.to.favicon.png.replace('{size}', size.toString()));
+            .toFile(paths.to.icon.png.replace('{size}', size.toString()));
     });
 
     return parallel(
-        jetpack.copyAsync(paths.from.favicon, paths.to.favicon.svg, { overwrite: true }),
+        jetpack.copyAsync(paths.from.icon, paths.to.icon.svg, { overwrite: true }),
         ...pngGeneration
     ) as unknown as Promise<void>;
-}, { inputs: [paths.from.favicon] });
+}, { inputs: [paths.from.icon] });
+
+const manifest = task('manifest', async () => {
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    let content = (await jetpack.readAsync(paths.from.manifest))!;

```

Comment:
Hmmm tbh I think I would name the manifest manifest.template.json and leave icon out completely. Read as json not parse line by line and add icon property in to json object

Reply:
I did something like that at first, but it made build a bit more complicated and manifest a bit less readable as you don't see the format straight away.

I could parse as JSON to find the template I suppose — I feel attached to specific formatting, but then it should be compressed in output anyways.

Reply:
Pretty much. Sounds good

Reply:
I tried and ended up with 
```typescript
    content.icons = content.icons.flatMap(icon => {
        if (icon.src.includes('{build:each-size}'))
            return [icon];

        const parts = Object.entries(icon);
        return iconSizes.map(size => {
            const replaced = parts.map(([key, value]) => [
                key, value.replace(/\{(?:build:each-)?size\}/g, size.toString())
            ]);
            return Object.fromEntries(replaced);
        });
    });
```

Which seems more complicated (tbh JS just needs better Object.map).

Reply:
Hmm
```typescript
    content.icons = content.icons.flatMap(icon => {
        if (icon.src.includes('{build:each-size}'))
            return [icon];

        return iconSizes.map(size => JSON.parse(
            JSON.stringify(icon).replace(/\{(?:build:each-)?size\}/g, size.toString())
        ) as typeof icon);
    });
```